encoding.go

```diff
@@ -11,17 +11,22 @@ const (
 	headersSizeInBytes   = timestampSizeInBytes + hashSizeInBytes + keySizeInBytes // Number of bytes used for all headers
 )
 
-func wrapEntry(timestamp uint64, hash uint64, key string, entry []byte) []byte {
-	var blob []byte
+func wrapEntry(timestamp uint64, hash uint64, key string, entry []byte, buffer *[]byte) []byte {
```

Comment:
I think it's a little bit to soon for this structure, I will happily use it if it will be used more than in one place to pass its values to func.